connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignorTest.java

```diff
@@ -220,19 +223,20 @@ public void testImmediateRevocationsWhenMaxDelayIs0()  {
         assertDelay(0);
         assertWorkers("worker1", "worker2", "worker3");
         assertConnectorAllocations(0, 1, 1);
-        assertTaskAllocations(2, 3, 3);
+        assertTaskAllocations(3, 3, 4);
```

Comment:
Curious, is this the result of the new Load Balancing assignment that you have come up with instead of the floor/ceiling based mechanism that was there before?

Reply:
Yep, it is! As long as it's safe to allocate them (which it should be in this case), it's better to see more connectors and tasks allocated instead of less, since that means that we're decreasing downtime while they're being moved across workers, and moving more quickly towards a balanced, stable assignment.

Reply:
Makes sense. This is great! Just a follow up question on `we're decreasing downtime while they're being moved across workers`, what are the advantages of moving more tasks v/s less? Because eventually in this test it converged to a balanced load. PS: I am not against this change at all but just for my understanding.

Reply:
The idea is that, if we've revoked some tasks, the sooner we reassign them, the better. There are some inefficiencies in the current allocation algorithm that cause tasks that were revoked in a prior round to not be reassigned in a future round, even when it would be perfectly safe to do so. This PR helps address some of those inefficiencies.

Of course, you're correct that in both cases, things converge to an even load, so in most cases, this optimization will be relatively minor.

Reply:
Oh. I wasn't aware of this (maybe missed it) => `There are some inefficiencies in the current allocation algorithm that cause tasks that were revoked in a prior round to not be reassigned in a future round, even when it would be perfectly safe to do so`. I think in this case it makes sense to do what you are suggesting.